spec/views/check_answers_page/client_content/assets_spec.rb

```diff
@@ -0,0 +1,183 @@
+require "rails_helper"
+
+RSpec.describe "estimates/check_answers.html.slim" do
+  let(:answers) { CheckAnswersPresenter.new(session_data) }
+
+  before do
+    assign(:answers, answers)
+    params[:id] = :id
+    allow(view).to receive(:form_with)
+    render template: "estimates/check_answers"
+  end
+
+  describe "client sections" do
+    context "when assets" do
+      context "when vehicle" do
+        let(:session_data) do
+          build(:minimal_session,
+                vehicle_owned:,
+                vehicle_value:,
+                vehicle_pcp:,
+                vehicle_finance:,
+                vehicle_over_3_years_ago:,
+                vehicle_in_regular_use:,
+                vehicle_in_dispute:)
+        end
+
+        context "when owns vehicle outright" do
+          let(:vehicle_owned) { true }
+          let(:vehicle_value) { 3_000 }
+          let(:vehicle_pcp) { false }
+          let(:vehicle_finance) { 0 }
+          let(:vehicle_over_3_years_ago) { false }
+          let(:vehicle_in_regular_use) { false }
+          let(:vehicle_in_dispute) { false }
+
+          it "renders content" do
+            expect(page_text).to include("Owns a vehicleYes")
+            expect(page_text).to include("Estimated value£3,000.00")
+            expect(page_text).to include("In regular useNo")
+            expect(page_text).to include("Bought over 3 years agoNo")
+            expect(page_text).to include("Payments left on vehicleNo")
+          end
+
+          context "when is smod" do
+            let(:vehicle_in_dispute) { true }
+
+            it "renders content" do
+              expect(page_text).to include("Disputed asset")
+            end
+          end
+        end
+
+        context "when owns a vehicle on finance" do
+          let(:vehicle_owned) { true }
+          let(:vehicle_value) { 2_000 }
+          let(:vehicle_pcp) { true }
+          let(:vehicle_finance) { 100 }
+          let(:vehicle_over_3_years_ago) { true }
+          let(:vehicle_in_regular_use) { true }
+          let(:vehicle_in_dispute) { false }
+
+          it "renders content correctly" do
+            expect(page_text).to include("Owns a vehicleYes")
```

Comment:
Oh, also, at the moment `page_text` is a functiona that does a bunch of work (parse an HTML string into a Nokogiri object, extract the text, smoosh it all together into an output string) every time it's called. That's going to slow these tests down unnecessarily. We should probably only call it once per test. Maybe by assigning its output to a local variable and then doing `expect` on that instead? (Although a lot of these multiple expects might go away if you do something like I suggested to expect these snippets of text to appear in a single order.)